### PR TITLE
Add validation to avoid fatal error on PHP 8 if index is not defined

### PR DESCRIPTION
**Pull request by yuloma**
Este cambio para corregir un error que le da a un cliente al hacer la indexación con PHP 8

Los detalles del error están en este ticket: https://github.com/doofinder/support/issues/2416

### CODE REVIEW

File: doofinder-for-woocommerce/includes/api/endpoints/class-endpoint-product.php [hunk 1]
```diff
@@ -634,23 +634,25 @@ private static function get_variations($products_data){
         $products = array();
 
         foreach($products_data as $product){
-            $type       = $product["type"];
-            $attributes = $product["attributes"];
+            if (!isset($product["type"])) {
```
Comment (davidmolinacano) [inline doofinder-for-woocommerce/includes/api/endpoints/class-endpoint-product.php:637]: It does not make sense for me. Maybe you wanted to include the opposite condition?

```suggestion
            if ( isset( $product["type"] ) ) {
```

**Review by davidmolinacano — CHANGES_REQUESTED** [review]
Also, try to increase the plugin version :smile:

File: doofinder-for-woocommerce/includes/api/endpoints/class-endpoint-product.php (hunk 1 above)
Comment (yuloma) [reply doofinder-for-woocommerce/includes/api/endpoints/class-endpoint-product.php:637]: You are right!

**Review by juan-gm — APPROVED** [review]

**Review by davidmolinacano — APPROVED** [review]
LGTM! 🙌

**Pull request merged**